src/main/kotlin/Main.kt

```diff
@@ -1,23 +1,12 @@
 import kotlinx.cli.*
-import kotlinx.cli.ArgParser
-import kotlinx.cli.ArgType
-import kotlinx.cli.required
-import kotlinx.coroutines.runBlocking
 
-fun main(args: Array<String>) = runBlocking() {
+@OptIn(ExperimentalCli::class)
+fun main(args: Array<String>) {
     val parser = ArgParser("kotlin-cli")
-    val startHeight by parser.option(ArgType.Int, shortName = "s", description = "Start Height").required()
-
+    val fetchBlockIdCommand = FetchBlockIdCommand()
+    val fetchTxIdsCommand = FetchTxIdsCommand()
+    parser.subcommands(fetchTxIdsCommand, fetchBlockIdCommand)
     parser.parse(args)
-
-    try {
-        val blockIds = fetchFirstBlockId(startHeight)
-        println(blockIds)
-
-        val txid = fetchTransactionId(blockIds)
-        txid.forEach { println(it) }
-    } catch (e: Exception) {
-        println("Error fetching block ID or transaction ID: $e")
-        kotlin.system.exitProcess(1)
-    }
-}
\ No newline at end of file
+    println(fetchBlockIdCommand.execute())
+    println(fetchTxIdsCommand.execute())
```

Comment:
I don't understand this, you shouldn't be printing that? You should be reacting to the arguments provided by the caller and calling the right command?

For example if I just run the CLI without any argument, I get a horrible error:

```
./gradlew run

> Task :run FAILED
Exception in thread "main" java.lang.IllegalStateException: Value for argument startHeight isn't set. ArgParser.parse(...) method should be called before.
        at kotlinx.cli.ArgumentSingleValue.getValue(ArgumentValues.kt:121)
        at kotlinx.cli.ArgumentValueDelegate$DefaultImpls.getValue(ArgParser.kt:59)
        at kotlinx.cli.ArgumentSingleValue.getValue(ArgumentValues.kt:118)
        at FetchBlockIdCommand.getStartHeight(FetchBlockIdCommand.kt:9)
        at FetchBlockIdCommand.access$getStartHeight(FetchBlockIdCommand.kt:5)
        at FetchBlockIdCommand$execute$1.invokeSuspend(FetchBlockIdCommand.kt:13)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
        at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
        at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
        at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
        at FetchBlockIdCommand.execute(FetchBlockIdCommand.kt:12)
        at MainKt.main(Main.kt:11)

```

Reply:
Earlier, the execution of each command was tied directly to the printing of results.
The results were directly printed to the console which made writing tests difficult for me.
I'd a doubt regarding it,(in the commit id  af0318c3245e1c74534f2775c5b86266b61ddc95 I was following this approach)

I later found that instead of directly printing the resuls, we can assign it to a variable and the output can be a string or something which will eventually make writing tests easier. That is the reason why I decided to refractor my execute() function.
Now, it's easier to test the execute() method directly by checking the returned result.

Let me know if the above approach is not correct.
The other alternatives use mockking the mempoolClient. I'm not very sure about this and need to look deeper here.

Reply:
See https://github.com/claddyk/MempoolCLI/pull/4#discussion_r1222713476